src/logistics/TerminalNetwork_v2.ts

```diff
@@ -1051,7 +1052,7 @@ export class TerminalNetworkV2 implements ITerminalNetwork {
 		for (const resource of RESOURCE_EXCHANGE_ORDER) {
 			for (const colony of (providers[resource] || [])) {
 				// Skip if the terminal is not ready -  prevents trying to send twice in a single tick
-				if (colony.terminal && !colony.terminal.isReady) {
+				if (!colony.terminal || !colony.terminal.isReady) {
```

Comment:
Is this something you encountered? IIRC a room without a terminal shouldn't be indexed in terminalNetwork. So if it's calling it here without a terminal we should probably catch it elsewhere.

Reply:
I thought that it was confusing before since if the terminal wasn't there, it would behave the same as if it was ready. If it's always there, then it shouldn't be checked, like: `if (!colony.terminal!.isReady) {`.